specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,503 @@
+    CoreWebView2FrameInfo mainFrameInfo = null;
+    while (frameInfo != null) {
+    firstLevelFrameInfo = mainFrameInfo;
+    mainFrameInfo = frameInfo;
+    frameInfo = frameInfo.ParentFrameInfo;
+    }
+    return firstLevelFrameInfo;
+}
+
+private async void ProcessFrameInfoCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    try
+    {
+        // <GetProcessInfosWithDetails>
+        IReadOnlyList<CoreWebView2ProcessInfo> processList = await webView.CoreWebView2.Environment.GetProcessInfosWithDetailsAsync();
+        int processCount = processList.Count;
+        string rendererProcessInfos = "";
+        string otherProcessInfos = "";
+        int rendererProcessCount = 0;
+        for (int i = 0; i < processCount; ++i)
+        {
+            CoreWebView2ProcessKind kind = processList[i].Kind;
+            int processId = processList[i].ProcessId;
+            if (kind == CoreWebView2ProcessKind.Renderer)
+            {
+                int frameInfoCount = 0;
+                string frameInfos = "";
+                // <AssociatedFrameInfos>
+                IReadOnlyList<CoreWebView2FrameInfo> frameInfoList = processList[i].AssociatedFrameInfos;
+                foreach (CoreWebView2FrameInfo frameInfo in frameInfoList)
+                {
+                    frameInfoCount++;
+                    frameInfos += AppendFrameInfo(frameInfo);
+                }
+                // </AssociatedFrameInfos>
+                string rendererProcessInfo = $"{frameInfoCount} frame info(s) found in renderer process ID: {processId}\n {frameInfos}";
+                rendererProcessInfos += $"{rendererProcessInfo} \n";
+                rendererProcessCount++;
+            }
+            else
+            {
+                otherProcessInfos += $"Process ID: {processId} | Process Kind: {kind}\n";
+            }
+        }
+        // </GetProcessInfosWithDetails>
+        string message = $"{processCount} process(es) found in total, from which {rendererProcessCount} renderer process(es) found\n\n" +
+                            $"{rendererProcessInfos}\nRemaining Process Infos:\n{otherProcessInfos}";
+        MessageBox.Show(this, message, "Process Info with Associated Frames");
+    }
+    catch (NotImplementedException exception)
+    {
+        MessageBox.Show(this, "GetProcessInfosWithDetailsAsync Failed: " + exception.Message,
+            "Process Info with Associated Frames");
+    }
+}
+```
+
+# API Details
+## C++
+```c++
+interface ICoreWebView2_18;
+interface ICoreWebView2Frame5;
+interface ICoreWebView2FrameInfo2;
+interface ICoreWebView2Environment14;
+interface ICoreWebView2ProcessInfo2;
+interface ICoreWebView2GetProcessInfosWithDetailsCompletedHandler;
+
+// Indicates the frame type used in the `ICoreWebView2FrameInfo2` interface.
+[v1_enum]
+typedef enum COREWEBVIEW2_FRAME_KIND {
+  /// Indicates that the frame is type of frame we don't differentiate yet.
+  COREWEBVIEW2_FRAME_KIND_OTHER,

```

Comment:
Do we ever report this for any other kind of frames beyond terminated frames? Could just call this `TERMINATED` and just use new values for future types of frames?

Calling it `OTHER` makes it hard to add new types of frames later. Suppose we add "3DFrames" later. Do they show up as `COREWEBVIEW2_FRAME_KIND_3DFRAME` and confuse old clients? Or do we report them as `COREWEBVIEW2_FRAME_KIND_OTHER` and confuse new clients?

Reply:
See above